### ISSUE TITLE
Batch Claim SLP

### ISSUE THREAD

**Issue by Joyen12** [FerranMarin/axie-scholar-utilities#23]
Amazing work because the info are so hard to find, they just have no doc...

Sorry if I am wrong but it seems you didn't implement batch SLP claim.

This is actually the most important. I guess maybe it's difficult to do ?

(I'm willing to help if you can put me in the right direction to do that)

Best regards

**Comment by FerranMarin** [comment]
I dobut that is possible, as claim has to be done in the axie infinity web. If you can think of a way... I am more than happy to try it.

**Comment by FerranMarin** [comment]
For this to happen, someone needs to figure out a way, I honestly doubt there is one.

**Comment by FerranMarin** [comment]
Closing this issue, if someone has an idea on how to do it... happy to re-open and implement it.

**Comment by AlvaroBro** [comment]
Hi @FerranMarin !!
First thanks for the amazing work!
I wanted to ask... is the claim functionality working? 
Reading this issue it seems it isn't, but I read in v1.0 changelog that this functionality was added back then. Did it stop working at some point or something like that? If so I'd like to help. What's exactly the current status?
I keep getting error 502 myself executing your client.

**Comment by FerranMarin** [comment]
v1.12.1 is currently the only working version of this code.
A part from the 2 branches.
So, yes, you should be able to make v1.12.1 work :)

**Comment by AlvaroBro** [comment]
> v1.12.1 is currently the only working version of this code. A part from the 2 branches. So, yes, you should be able to make v1.12.1 work :)

Great! Is it possible that server returns status code `502` for the claim request if there is no claimable SLP?
thanks!

**Comment by FerranMarin** [comment]
Nope.
`502` HTTP error code means a server error...

**Comment by AlvaroBro** [comment]
Ok then I don't know what is going on, I keep getting that `502`.

```
run python axie_scholar_cli.py claim ../../sample_data/payments.json ../../sample_data/secrets.json
2021-11-25 18:22:51,054 - INFO: I shall claim SLP
2021-11-25 18:22:51,055 - INFO: Secret file correctly validated
2021-11-25 18:22:51,059 - INFO: Claiming starting...
2021-11-25 18:22:51,492 - INFO: Account Account #2 (ronin:...) has 11 unclaimed SLP
2021-11-25 18:22:52,488 - INFO: Important: Claim for account Account #2 (ronin:...) had to be skipped (502)
2021-11-25 18:22:52,489 - INFO: Claiming completed!
```

**Comment by FerranMarin** [comment]
Amm... why are you using sample data files and expect it to claim? On another note, this could be you have a wrong private key in secrets.json or any other thing that is causing the SM server to respond with a 502 error.

All I can say, check and make sure you using the latest version, make sure your data in payments and secrets is correct.
Also, come to discord for easier communication and trouble shooting.

**Comment by AlvaroBro** [comment]
I created those files, I just stored in sample_data folder. As far as I know they are correct, they have the minimum info and I used your clients tool to check the secrets file. 
Also, would the key or account be incorrect, then `get_jwt` would fail, and it does not fail as getting the token is a prior step to claiming... :(

Anyways, I keep trying. I just joined Discord as well, any ideas are welcome :)

**secrets.json**

```
{
    "ronin:0dd...": "0x..."
}
```

**payments.json**

```
{
    "Scholars": [
        {
            "Name": "Account #2",
            "AccountAddress": "ronin:0dd..."
        }
    ]
}
```
